// originate-react-components-library/src/components/Tag/Tag.js
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rem } from 'polished';

const Tag = ({ children, className, status }) => {
  return (
    <StyledTag className={` ${status == null ? '' : status}`}>
      {status}
    </StyledTag>
  );
};

Tag.propTypes = {
  status: PropTypes.string
};

const StyledTag = styled.span`
  background: ${({ theme }) => theme.__OriginateLib.colors.text};
  border-radius: ${rem('4px')};
  color: ${({ theme }) => theme.__OriginateLib.colors.white};
  display: inline-block;
  font-size: ${rem('12px')};
  line-height: ${rem('24px')};
  padding: 0 ${rem('15px')};
  max-width: 100%;
  text-align: center;
  text-transform: capitalize;

  &.paid,
  &.green,
  &.success,
  &.approved,
  &.disbursed,
  &.confirmed,
  &.successful {
    background: ${({ theme }) => theme.__OriginateLib.colors.green};
  }

  &.red,
  &.denied,
  &.failed,
  &.failure,
  &.declined,
  &.unresolved,
  &.unconfirmed {
    background: ${({ theme }) => theme.__OriginateLib.colors.red};
  }

  &.yellow,
  &.pending,
  &.on-hold,
  &.waiting,
  &.review,
  &.in-progress {
    background: ${({ theme }) => theme.__OriginateLib.colors.yellow};
  }
`;

export { Tag };
