// originate-react-components-library/src/components/Card/Card.js
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rem } from 'polished';

const Card = ({ title, children, subtitle, className }) => {
  return (
    <StyledCard className={className}>
      <div className="card-desc">
        {title == null ? (
          ''
        ) : (
          <Fragment>
            <CardTitle>{title}</CardTitle>
            <CardSubtitle>{subtitle}</CardSubtitle>
          </Fragment>
        )}
      </div>
      <CardContent>{children}</CardContent>
    </StyledCard>
  );
};

const CardTitle = styled.h4`
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
  margin: 0 0 ${rem('5px')};
`;
const CardSubtitle = styled.h5`
  color: ${({ theme }) => theme.__OriginateLib.colors.text2};
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.regular};
  margin: 0 0;
`;

const CardContent = styled.div`
  max-width: 500px;
`;

const StyledCard = styled.div`
  display: grid;
  grid-gap: ${rem('50px')};
  grid-template-columns: ${rem('300px')} 1fr;
  margin: 0 0 ${rem('50px')};

  ${'' /* @include phone {
    grid-template-columns: 1fr;
  } */}
`;

Card.propTypes = {
  title: PropTypes.string,
  classes: PropTypes.string,
  subtitle: PropTypes.string
};

export default Card;
