// originate-react-components-library/src/components/Popup/Popup.js
import React from 'react';
import PropTypes from 'prop-types';
import { DataCard } from '../index';
import styled from 'styled-components';
import { rgba, rem } from 'polished';
import img from '../../static/icons/checklist.svg';

const Popup = ({ className, title, children }) => {
  return (
    <StyledPopup className={`${className == null ? '' : className}`}>
      <StyledContent>
        <DataCard title={title}>{children}</DataCard>
      </StyledContent>
    </StyledPopup>
  );
};

const StyledPopup = styled.section`
  height: 100vh;
  left: 0;
  padding: 0 15px;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 99;

  ul {
    margin: 0 0 ${rem('50px')};
    padding: 0;

    li {
      line-height: ${rem('25px')};
      list-style: none;
      margin: 0 0 ${rem('15px')};

      @include phone {
        margin: 0 0 ${rem('10px')};
      }

      &:before {
        background: url(${img}) center left no-repeat;
        border-radius: 0%;
        color: ${({ theme }) => theme.__OriginateLib.colors.white};
        content: '';
        display: inline-block;
        float: left;
        font-size: ${rem('15px')};
        font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
        height: ${rem('25px')};
        line-height: ${rem('33px')};
        margin-right: ${rem('10px')};
        text-align: center;
        width: ${rem('25px')};
      }
    }
  }
  .button-list {
    display: grid;
    grid-template-columns: 1fr;
    text-align: center;

    .button {
      margin: 0 auto;
      max-width: 100%;
      width: 300px;
    }

    .link {
      color: ${({ theme }) => rgba(theme.__OriginateLib.colors.text, 0.5)};
      margin: ${rem('15px')} auto;
    }
  }
  &:before {
    background: ${({ theme }) => rgba(theme.__OriginateLib.colors.black, 0.4)};
    bottom: 0;
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    width: 100%;
    z-index: -1;
  }
`;
const StyledContent = styled.div`
  border-radius: 1.2rem;
  margin: 0 auto;
  max-height: 90vh;
  max-width: ${rem('550px')};
  overflow: auto;
  position: relative;
  top: 50%;
  transform: translateY(-50%);
  width: 100%;

  &.loan-requirements {
    .data-card {
      margin: 0;
      padding: ${rem('50px')};

      @include phone {
        padding: ${rem('20px')};
      }
    }
    &__title {
      margin-left: auto;
      margin-right: auto;
      max-width: ${rem('352px')};
      text-align: center;
    }

    &__content {
      margin-left: auto;
      margin-right: auto;
      max-width: ${rem('352px')};
    }
    p {
      font-size: ${rem('20px')};
      margin: 0 0 ${rem('30px')};
    }
  }
`;

Popup.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string
};

export default Popup;
