// originate-react-components-library/src/components/DebitCard/DebitCard.js
import React from 'react';
import Radio from '../FormElements/Radio/Radio';
import Ellipsis from '../Icon/Ellipsis';
import styled from 'styled-components';
import { rgba, rem } from 'polished';
import img from '../../static/icons/check-icon.svg';
import { Ellipse } from '../Icon/Ellipse';
import { MasterCard } from '../Icon';
import PropTypes from 'prop-types';

const DebitCard = ({
  isDefault = false,
  className,
  cardNumber,
  expiryDate
}) => {
  return (
    <StyledDebitCard className={className}>
      <CardTop>
        <CardStatus>
          <Radio defaultChecked={isDefault} />
        </CardStatus>
        <CardIcon>
          <Ellipsis />
        </CardIcon>
      </CardTop>
      <CardMid>
        <span>
          <Ellipse />
        </span>
        <span>
          <Ellipse />
        </span>
        <span>
          <Ellipse />
        </span>
        <CardNumber>{cardNumber}</CardNumber>
      </CardMid>
      <CardBottom>
        <CardExpiry>
          <CardExpires>Expires:</CardExpires>{' '}
          <CardExpiryDate>{expiryDate}</CardExpiryDate>
        </CardExpiry>
        <CardType>
          <MasterCard />
        </CardType>
      </CardBottom>
    </StyledDebitCard>
  );
};
const CardTop = styled.div`
  margin-bottom: ${rem('48px')};
`;

const CardStatus = styled.div`
  display: inline-block;
`;

const CardIcon = styled.div`
  display: inline;
  float: right;
`;

const CardMid = styled.div`
  align-content: space-evenly;
  display: flex;
  flex-wrap: wrap;
  margin-bottom: ${rem('18px')};

  span {
    line-height: ${rem('21px')};
  }
`;
const CardNumber = styled.span`
  font-size: ${rem('18px')};
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
  line-height: ${rem('28px')};

  ${'' /* @include phone {
    font-size: ${rem('14px')};
  } */}
`;

const CardBottom = styled.div`
  display: inline;
`;

const CardExpiry = styled.div`
  float: left;
  margin-top: ${rem('45px')};
`;

const CardExpires = styled.span`
  font-size: ${rem('14px')};
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
  line-height: ${rem('28px')};
`;

const CardExpiryDate = styled.span`
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
`;

const CardType = styled.div`
  float: right;
  margin-top: ${rem('45px')};
`;

const StyledDebitCard = styled.div`
  background: ${({ theme }) => theme.__OriginateLib.colors.primary};
  order: ${rem('1px')} solid #d2d6db;
  border-radius: ${rem('12px')};
  box-shadow: ${rem('2px')} ${rem('11px')} ${rem('22px')} ${rgba(
  0,
  0,
  0,
  0.06
)};
  color: ${({ theme }) => theme.__OriginateLib.colors.white};
  display: block;
  height: ${rem('186px')};
  max-width: 100%;
  padding: ${rem('20px')};
  width: ${rem('345px')};

  ${'' /* include phone {
      width: 100%;
    } */}

  ${'' /* .radio {
  } */}


  .radio {
    margin: 0;
    input {
      &:checked + span {
        background: none;
        border-color: transparent;
        color: transparent;
        box-shadow: none;

        &:before {
          background: none;
          border: ${rem('2px')} solid
            ${({ theme }) => theme.__OriginateLib.colors.white};
          box-sizing: border-box;
        }

        &:after {
          background: ${({ theme }) => theme.__OriginateLib.colors.secondary}
            url(${img}) center no-repeat;
        }
      }
    }

    span {
      border: none;
      float: none;

      &:before {
        border: ${rem('2px')} solid
          ${({ theme }) => theme.__OriginateLib.colors.white};
        box-shadow: ${rem('0px')} ${rem('4px')} ${rem('4px')}
          ${({ theme }) => rgba(0, 0, 0, 0.02)};
        height: ${rem('25px')};
        top: 0;
        width: ${rem('25px')};
        background: none;
      }

      &:after {
        width: ${rem('21px')};
        top: ${rem('2px')};
        height: ${rem('21px')};
        left: 12px;
      }
    }
  
`;
DebitCard.propTypes = {
  isDefault: PropTypes.bool,
  className: PropTypes.string,
  cardNumber: PropTypes.string,
  expiryDate: PropTypes.string
};
DebitCard.defaultProps = {
  className: '',
  isDefault: true,
  cardNumber: '',
  expiryDate: ''
};
export default DebitCard;
