// originate-react-components-library/src/components/RepaymentBreakdown/RepaymentBreakdown.js
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rgba, rem } from 'polished';

const RepaymentBreakdown = ({ children }) => {
  return (
    <StyledRepaymentBreakdown className="">{children}</StyledRepaymentBreakdown>
  );
};
const StyledRepaymentBreakdown = styled.ul`
  background: ${({ theme }) =>
    rgba(theme.__OriginateLib.colors.secondary, 0.1)};
  border: 1px solid ${({ theme }) => theme.__OriginateLib.colors.secondary};
  border-radius: 4px;
  font-size: ${rem('16px')};
  margin: 0;
  padding: ${rem('25px')};

  ${'' /* @include phone {
    font-size: 15px;
    padding: 10px;
  } */}

  li {
    font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
    list-style: none;
    line-height: ${rem('20px')};
    margin: 0 0 t ${rem('15px')};

    @include phone {
      margin: 0 0 ${rem('10px')};
    }

    span.value {
      float: right;
    }

    &:last-of-type {
      margin: 0;
    }
  }

  .note {
    text-align: center;
  }
`;
RepaymentBreakdown.propTypes = {
  className: PropTypes.string
};

export default RepaymentBreakdown;
