// originate-react-components-library/src/stylesheets/Global.js
import { createGlobalStyle } from 'styled-components';
import { rem } from 'polished';

const GlobalStyles = createGlobalStyle`

html, body{
    background: #FCFEFA;
    font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.regular};
    min-height: 100%;
}

body, input, textarea, select, button {
    color:  ${({ theme }) => theme.__OriginateLib.colors.text};
    font-family: ${({ theme }) => theme.__OriginateLib.fontFamily.default};
        font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.regular};
    font-size: ${rem('14px')};
    line-height: ${rem('24px')}
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

a, a:focus, a:hover {
	color: inherit;
	display: inline-block;
	text-decoration: none;
	-webkit-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
	-moz-transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
	transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s
}
img{
	max-width: 100%;
}

.margin-top{
	margin-top: ${rem('30px')};

}
h1, h2, h3, h4, h5, h6 {
	margin-top: 0;
}

h1{
	font-size: ${rem('44px')};
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
	line-height: ${rem('64px')};
}

h2{
	font-size: ${rem('32px')};
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
	line-height: ${rem('48px')};
}

h3{
	font-size: ${rem('24px')};
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
	line-height: ${rem('32px')};
}

h4{
	font-size: ${rem('20px')};
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
	line-height: ${rem('27px')};
	color:${({ theme }) => theme.__OriginateLib.colors.text3};
}

h5{
	font-size: ${rem('16px')};
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
	line-height: ${rem('24px')};
}

h6{
	font-size: ${rem('14px')};
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
	line-height: ${rem('27px')};
	color:${({ theme }) => theme.__OriginateLib.colors.text3};
}

::-moz-selection {
	background: ${({ theme }) => theme.__OriginateLib.colors.primary};
	color: #ffffff;
	opacity: 1;
	text-shadow: none;
}

::selection {
	background: ${({ theme }) => theme.__OriginateLib.colors.primary};
	color: #ffffff;
	opacity: 1;
	text-shadow: none;
}

.center-text{
	text-align: center;
}

.left-text{
	text-align: left;
}

.right-text{
	text-align: right;
}

.bold-text{
	font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.bold};
}

.no-loan{
	color:${({ theme }) => theme.__OriginateLib.colors.text2};
	text-align: center;

	h3{
		font-size: ${rem('28px')};
		font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.regular};
		line-height:  ${rem('35px')};
		margin: 0 auto  ${rem('40px')};
		max-width:  ${rem('500px')};
	}

	.smiley-holder{
		margin: 0 0  ${rem('30px')};

		svg{
			width: 45px;
		}

	}
}
.hidden {
	opacity: 0;
}

.resend-otp,
.resend-verification{
	color:  ${({ theme }) => theme.__OriginateLib.colors.secondary};
	cursor: pointer;
	font-weight:  ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};;
	text-decoration: underline;
	transition: 0.3s;

	&:hover{
		text-decoration: none;
	}
}
`;

export { GlobalStyles };
