// originate-react-components-library/src/components/BankCard/BankCard.js
import Ellipsis from '../Icon/Ellipsis';
import React from 'react';
import styled from 'styled-components';
import Radio from '../FormElements/Radio/Radio';
import { rgba, rem } from 'polished';
import img from '../../static/icons/check-icon.svg';
import PropTypes from 'prop-types';

const BankCard = ({
  className,
  isDefault,
  accountNumber,
  accountName,
  bankName,
  style
}) => {
  return (
    <StyledBankCard className={className} style={style}>
      <CardTop>
        <CardStatus>
          <Radio defaultChecked={isDefault} />
        </CardStatus>
        <CardIcon>
          <Ellipsis />
        </CardIcon>
      </CardTop>
      <div>
        <AccountNumber>{accountNumber}</AccountNumber>
        <Bank>
          <AccountName>{accountName}</AccountName>
          <BankName>{bankName}</BankName>
        </Bank>
      </div>
    </StyledBankCard>
  );
};

const CardTop = styled.div`
  margin-bottom: ${rem('18px')};
`;

const CardStatus = styled.div`
  display: inline-block;
`;

const CardIcon = styled.div`
  display: inline;
`;

const AccountNumber = styled.div`
  font-size: ${rem('18px')};
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
  line-height: ${rem('28px')};

  ${'' /* @include phone {
    font-size: ${rem('14px')};
  } */}
`;

const Bank = styled.div`
  display: inline;
  font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
  line-height: ${rem('28px')};
`;

const AccountName = styled.div`
  float: left;
  font-size: ${rem('18px')};

  ${'' /* @include phone {
    font-size: ${rem('14px')};
  } */}
`;

const BankName = styled.div`
  float: right;
  font-size: ${rem('14px')};
`;

const StyledBankCard = styled.div`
  background: linear-gradient(
    219.38deg,
    ${rgba(240, 247, 255, 0)},
    21.46%,
    #f0f7ff 77.46%
  );
  border: ${rem('1px')} solid #d2d6db;
  border-radius: ${rem('12px')};
  box-shadow: ${rem('2px')} ${rem('11px')} ${rem('22px')} ${rgba(
  0,
  0,
  0,
  0.06
)};
  color: ${({ theme }) => theme.__OriginateLib.colors.black};
  display: block;
  height: ${rem('130px')};
  max-width: 100%;
  padding: ${rem('20px')};
  width: ${rem('345px')};

  ${'' /* @include phone {
    width: 100%;
  } */}

  .radio {
    margin: 0;

    input {
      &:checked + span {
        background: none;
        border-color: transparent;
        color: transparent;
        box-shadow: none;

        &:before {
          border: 2px solid ${({ theme }) => theme.__OriginateLib.colors.white};
          box-sizing: border-box;
          background: none;
        }

        &:after {
          background: ${({ theme }) => theme.__OriginateLib.colors.secondary}
            url(${img}) center no-repeat;
        }
      }
    }

    span {
      border: none;
      float: none;

      &:before {
        border: ${rem('2px')} solid
          ${({ theme }) => theme.__OriginateLib.colors.white};
        box-shadow: ${rem('0px')} ${rem('4px')} ${rem('4px')}
          ${rgba(0, 0, 0, 0.02)};
        height: ${rem('25px')};
        top: 0;
        width: ${rem('25px')};
        background: none;
      }

      &:after {
        width: ${rem('21px')};
        top: ${rem('2px')};
        height: ${rem('21px')};
        left: 12px;
      }
    }
  
`;

BankCard.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  isDefault: PropTypes.bool,
  accountNumber: PropTypes.string,
  accountName: PropTypes.string,
  bankName: PropTypes.string
};
export default BankCard;
