// originate-react-components-library/src/components/FormElements/Radio/Radio.js
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rem, rgba } from 'polished';

const Radio = ({
  className,
  value,
  disabled,
  checked,
  defaultChecked,
  readOnly,
  name,
  placeholder,
  onChange,
  label
}) => {
  return (
    <StyledRadio className="radio">
      <input
        className={`${className == null ? '' : className}`}
        type="radio"
        value={value}
        disabled={disabled}
        checked={checked}
        defaultChecked={defaultChecked}
        readOnly={readOnly}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
      />
      <span>{label}</span>
    </StyledRadio>
  );
};

const StyledRadio = styled.label`
  cursor: pointer;
  display: inline-block;
  margin: 0 ${rem('15px')} ${rem('20px')} 0;

  input {
    cursor: pointer;
    position: absolute;
    margin: 0;
    width: 0;
    height: 0;
    opacity: 0;

    &:checked + span {
      background: ${({ theme }) => theme.__OriginateLib.colors.white};
      border-color: ${({ theme }) => theme.__OriginateLib.colors.secondary};
      color: ${({ theme }) => theme.__OriginateLib.colors.primary};
      box-shadow: 0 3px 15px
        ${({ theme }) => rgba(theme.__OriginateLib.colors.secondary, 0.5)};

      &:before {
        border-color: ${({ theme }) => theme.__OriginateLib.colors.secondary};
        background: ${({ theme }) => theme.__OriginateLib.colors.secondary};
        box-shadow: 0 3px 15px ${({ theme }) => rgba(157, 216, 88, 0.5)};
      }
      &:after {
        background: ${({ theme }) => theme.__OriginateLib.colors.white};
      }
    }
  }

  span {
    border: 1px solid ${({ theme }) => theme.__OriginateLib.colors.text};
    border-radius: 4px;
    cursor: pointer;
    display: block;
    font-weight: ${({ theme }) => theme.__OriginateLib.fontWeight.semiBold};
    float: left;
    line-height: ${rem('20px')};
    margin: 0;
    outline: none;
    padding: ${rem('10px')} ${rem('10px')} ${rem('10px')} ${rem('40px')};
    position: relative;
    transition: all 0.2s ease-in;
    width: 100%;

    &:before {
      border: 1px solid ${({ theme }) => theme.__OriginateLib.colors.text};
      border-radius: 100%;
      box-sizing: border-box;
      content: '';
      left: 10px;
      top: 10px;
      height: 20px;
      position: absolute;
      width: 20px;
      background: transparent;
    }

    &:after {
      border-radius: 100%;
      box-sizing: border-box;
      content: '';
      left: ${rem('16px')};
      top: ${rem('16px')};
      height: 8px;
      position: absolute;
      width: 8px;
    }
  }
`;

Radio.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  classes: PropTypes.string,
  placeholder: PropTypes.string,
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  checked: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  readOnly: PropTypes.oneOfType([PropTypes.bool, PropTypes.string])
};

export default Radio;
