// originate-react-components-library/dist/components/BankCard/BankCard.js
function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n  background: linear-gradient(\n    219.38deg,\n    ", ",\n    21.46%,\n    #f0f7ff 77.46%\n  );\n  border: ", " solid #d2d6db;\n  border-radius: ", ";\n  box-shadow: ", " ", " ", " ", ";\n  color: ", ";\n  display: block;\n  height: ", ";\n  max-width: 100%;\n  padding: ", ";\n  width: ", ";\n\n  ", "\n\n  .radio {\n    margin: 0;\n\n    input {\n      &:checked + span {\n        background: none;\n        border-color: transparent;\n        color: transparent;\n        box-shadow: none;\n\n        &:before {\n          border: 2px solid ", ";\n          box-sizing: border-box;\n          background: none;\n        }\n\n        &:after {\n          background: ", "\n            url(", ") center no-repeat;\n        }\n      }\n    }\n\n    span {\n      border: none;\n      float: none;\n\n      &:before {\n        border: ", " solid\n          ", ";\n        box-shadow: ", " ", " ", "\n          ", ";\n        height: ", ";\n        top: 0;\n        width: ", ";\n        background: none;\n      }\n\n      &:after {\n        width: ", ";\n        top: ", ";\n        height: ", ";\n        left: 12px;\n      }\n    }\n  \n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n  float: right;\n  font-size: ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  float: left;\n  font-size: ", ";\n\n  ", "\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  display: inline;\n  font-weight: ", ";\n  line-height: ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  font-size: ", ";\n  font-weight: ", ";\n  line-height: ", ";\n\n  ", "\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  display: inline;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: inline-block;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  margin-bottom: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import Ellipsis from '../Icon/Ellipsis';
import React from 'react';
import styled from 'styled-components';
import Radio from '../FormElements/Radio/Radio';
import { rgba, rem } from 'polished';
import img from '../../static/icons/check-icon.svg';
import PropTypes from 'prop-types';

var BankCard = function BankCard(_ref) {
  var className = _ref.className,
      isDefault = _ref.isDefault,
      accountNumber = _ref.accountNumber,
      accountName = _ref.accountName,
      bankName = _ref.bankName,
      style = _ref.style;
  return React.createElement(StyledBankCard, {
    className: className,
    style: style
  }, React.createElement(CardTop, null, React.createElement(CardStatus, null, React.createElement(Radio, {
    defaultChecked: isDefault
  })), React.createElement(CardIcon, null, React.createElement(Ellipsis, null))), React.createElement("div", null, React.createElement(AccountNumber, null, accountNumber), React.createElement(Bank, null, React.createElement(AccountName, null, accountName), React.createElement(BankName, null, bankName))));
};

var CardTop = styled.div(_templateObject(), rem('18px'));
var CardStatus = styled.div(_templateObject2());
var CardIcon = styled.div(_templateObject3());
var AccountNumber = styled.div(_templateObject4(), rem('18px'), function (_ref2) {
  var theme = _ref2.theme;
  return theme.__OriginateLib.fontWeight.semiBold;
}, rem('28px'), ''
/* @include phone {
font-size: ${rem('14px')};
} */
);
var Bank = styled.div(_templateObject5(), function (_ref3) {
  var theme = _ref3.theme;
  return theme.__OriginateLib.fontWeight.semiBold;
}, rem('28px'));
var AccountName = styled.div(_templateObject6(), rem('18px'), ''
/* @include phone {
font-size: ${rem('14px')};
} */
);
var BankName = styled.div(_templateObject7(), rem('14px'));
var StyledBankCard = styled.div(_templateObject8(), rgba(240, 247, 255, 0), rem('1px'), rem('12px'), rem('2px'), rem('11px'), rem('22px'), rgba(0, 0, 0, 0.06), function (_ref4) {
  var theme = _ref4.theme;
  return theme.__OriginateLib.colors.black;
}, rem('130px'), rem('20px'), rem('345px'), ''
/* @include phone {
width: 100%;
} */
, function (_ref5) {
  var theme = _ref5.theme;
  return theme.__OriginateLib.colors.white;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.__OriginateLib.colors.secondary;
}, img, rem('2px'), function (_ref7) {
  var theme = _ref7.theme;
  return theme.__OriginateLib.colors.white;
}, rem('0px'), rem('4px'), rem('4px'), rgba(0, 0, 0, 0.02), rem('25px'), rem('25px'), rem('21px'), rem('2px'), rem('21px'));
BankCard.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  isDefault: PropTypes.bool,
  accountNumber: PropTypes.string,
  accountName: PropTypes.string,
  bankName: PropTypes.string
};
export default BankCard;
