// originate-react-components-library/dist/components/FormElements/Radio/Radio.js
function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  cursor: pointer;\n  display: inline-block;\n  margin: 0 ", " ", " 0;\n\n  input {\n    cursor: pointer;\n    position: absolute;\n    margin: 0;\n    width: 0;\n    height: 0;\n    opacity: 0;\n\n    &:checked + span {\n      background: ", ";\n      border-color: ", ";\n      color: ", ";\n      box-shadow: 0 3px 15px\n        ", ";\n\n      &:before {\n        border-color: ", ";\n        background: ", ";\n        box-shadow: 0 3px 15px ", ";\n      }\n      &:after {\n        background: ", ";\n      }\n    }\n  }\n\n  span {\n    border: 1px solid ", ";\n    border-radius: 4px;\n    cursor: pointer;\n    display: block;\n    font-weight: ", ";\n    float: left;\n    line-height: ", ";\n    margin: 0;\n    outline: none;\n    padding: ", " ", " ", " ", ";\n    position: relative;\n    transition: all 0.2s ease-in;\n    width: 100%;\n\n    &:before {\n      border: 1px solid ", ";\n      border-radius: 100%;\n      box-sizing: border-box;\n      content: '';\n      left: 10px;\n      top: 10px;\n      height: 20px;\n      position: absolute;\n      width: 20px;\n      background: transparent;\n    }\n\n    &:after {\n      border-radius: 100%;\n      box-sizing: border-box;\n      content: '';\n      left: ", ";\n      top: ", ";\n      height: 8px;\n      position: absolute;\n      width: 8px;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rem, rgba } from 'polished';

var Radio = function Radio(_ref) {
  var className = _ref.className,
      value = _ref.value,
      disabled = _ref.disabled,
      checked = _ref.checked,
      defaultChecked = _ref.defaultChecked,
      readOnly = _ref.readOnly,
      name = _ref.name,
      placeholder = _ref.placeholder,
      onChange = _ref.onChange,
      label = _ref.label;
  return React.createElement(StyledRadio, {
    className: "radio"
  }, React.createElement("input", {
    className: "".concat(className == null ? '' : className),
    type: "radio",
    value: value,
    disabled: disabled,
    checked: checked,
    defaultChecked: defaultChecked,
    readOnly: readOnly,
    name: name,
    placeholder: placeholder,
    onChange: onChange
  }), React.createElement("span", null, label));
};

var StyledRadio = styled.label(_templateObject(), rem('15px'), rem('20px'), function (_ref2) {
  var theme = _ref2.theme;
  return theme.__OriginateLib.colors.white;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.__OriginateLib.colors.secondary;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.__OriginateLib.colors.primary;
}, function (_ref5) {
  var theme = _ref5.theme;
  return rgba(theme.__OriginateLib.colors.secondary, 0.5);
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.__OriginateLib.colors.secondary;
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.__OriginateLib.colors.secondary;
}, function (_ref8) {
  var theme = _ref8.theme;
  return rgba(157, 216, 88, 0.5);
}, function (_ref9) {
  var theme = _ref9.theme;
  return theme.__OriginateLib.colors.white;
}, function (_ref10) {
  var theme = _ref10.theme;
  return theme.__OriginateLib.colors.text;
}, function (_ref11) {
  var theme = _ref11.theme;
  return theme.__OriginateLib.fontWeight.semiBold;
}, rem('20px'), rem('10px'), rem('10px'), rem('10px'), rem('40px'), function (_ref12) {
  var theme = _ref12.theme;
  return theme.__OriginateLib.colors.text;
}, rem('16px'), rem('16px'));
Radio.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  classes: PropTypes.string,
  placeholder: PropTypes.string,
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  checked: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  readOnly: PropTypes.oneOfType([PropTypes.bool, PropTypes.string])
};
export default Radio;
