// originate-react-components-library/dist/components/DataCard/DataCard.js
function _templateObject5() {
  var data = _taggedTemplateLiteral([""]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  margin: 0 0 ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n      &:before {\n        background: ", ";\n        border-radius: ", ";\n        cursor: wait;\n        content: '';\n        height: 100%;\n        position: absolute;\n        left: 0;\n        top: 0;\n        z-index: 9;\n        width: 100%;\n      }\n\n      &:after {\n        animation: ", " 0.75s linear infinite;\n        border: 7.5px solid\n          ", ";\n        border-radius: 50%;\n        border-right-color: ", ";\n        box-sizing: border-box;\n        content: '';\n        display: inline-block;\n        height: ", ";\n        left: 0;\n        margin: auto;\n        top: calc(50% - 37.5px);\n        position: absolute;\n        right: 0;\n        width: ", ";\n        z-index: 99;\n      }\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n\tborder-radius: ", ";\n\tbox-shadow: 0 4px 8px ", ";\n\tmargin: 0 0 ", ";\n\tpadding: ", ";\n\tposition: relative;\n\tbackground: #FFFFFF;\n\t\n\t", "\n\n\t&.no-pad {\n\t\tpadding: 0;\n  }\n  \n  &.small {\n\t\twidth: 350px;\n\t}\n\n\t&.medium {\n\t\twidth: 600px;\n\t}\n\n\t&.large {\n\t\twidth: 900px;\n\t}\n\n\t&.custom {\n\t\twidth: 100%;\n\t}\n\t\n\t", "\n\n    &:after {\n      animation: ", " 0.75s linear infinite;\n      border: 7.5px solid\n        ", ";\n      border-radius: 50%;\n      border-right-color: ", ";\n      box-sizing: border-box;\n      content: '';\n      display: inline-block;\n      /* float: left; */\n      height: ", ";\n      left: 0;\n      margin: auto;\n      top: calc(50% - 37.5px);\n      position: absolute;\n      right: 0;\n      width: ", ";\n      z-index: 99;\n    }\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n\tto {\n      transform: rotate(1turn);\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styled, { keyframes, css } from 'styled-components';
import { rgba, rem } from 'polished';

var DataCard = function DataCard(_ref) {
  var title = _ref.title,
      children = _ref.children,
      noPadding = _ref.noPadding,
      loading = _ref.loading,
      className = _ref.className,
      size = _ref.size,
      style = _ref.style;
  return React.createElement(StyledDataCard, {
    noPadding: noPadding,
    loading: loading,
    className: classnames("".concat(size, " ").concat(className, " ").concat(loading === true ? 'loading' : '')),
    style: style
  }, title && React.createElement(StyledTitle, null, title), React.createElement(StyledContent, null, children));
};

var spin = keyframes(_templateObject());
var StyledDataCard = styled.div(_templateObject2(), rem('12px'), rgba(0, 0, 0, 0.1), rem('32px'), function (_ref2) {
  var noPadding = _ref2.noPadding;
  return noPadding ? '0' : "".concat(rem('40px'), " ").concat(rem('32px'));
}, ''
/* @include phone {
padding: ${rem('25px')};
} */
, function (_ref3) {
  var loading = _ref3.loading;
  return loading && css(_templateObject3(), function (_ref4) {
    var theme = _ref4.theme;
    return rgba(theme.__OriginateLib.colors.white, 0.75);
  }, rem('12px'), spin, function (_ref5) {
    var theme = _ref5.theme;
    return rgba(theme.__OriginateLib.colors.primary, 0.25);
  }, function (_ref6) {
    var theme = _ref6.theme;
    return theme.__OriginateLib.colors.primary;
  }, rem('75px'), rem('75px'));
}, spin, function (_ref7) {
  var theme = _ref7.theme;
  return rgba(theme.__OriginateLib.colors.primary, 0.25);
}, function (_ref8) {
  var theme = _ref8.theme;
  return theme.__OriginateLib.colors.primary;
}, rem('75px'), rem('75px'));
var StyledTitle = styled.h4(_templateObject4(), rem('24px'));
var StyledContent = styled.div(_templateObject5());
DataCard.propTypes = {
  title: PropTypes.string,
  noPad: PropTypes.bool,
  loading: PropTypes.bool,
  className: PropTypes.string,
  size: PropTypes.string,
  style: PropTypes.object
};
DataCard.defaultProps = {
  className: '',
  title: '',
  noPad: false,
  size: 'custom',
  style: {}
};
export default DataCard;
