// originate-react-components-library/dist/components/DebitCard/DebitCard.js
function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n  background: ", ";\n  order: ", " solid #d2d6db;\n  border-radius: ", ";\n  box-shadow: ", " ", " ", " ", ";\n  color: ", ";\n  display: block;\n  height: ", ";\n  max-width: 100%;\n  padding: ", ";\n  width: ", ";\n\n  ", "\n\n  ", "\n\n\n  .radio {\n    margin: 0;\n    input {\n      &:checked + span {\n        background: none;\n        border-color: transparent;\n        color: transparent;\n        box-shadow: none;\n\n        &:before {\n          background: none;\n          border: ", " solid\n            ", ";\n          box-sizing: border-box;\n        }\n\n        &:after {\n          background: ", "\n            url(", ") center no-repeat;\n        }\n      }\n    }\n\n    span {\n      border: none;\n      float: none;\n\n      &:before {\n        border: ", " solid\n          ", ";\n        box-shadow: ", " ", " ", "\n          ", ";\n        height: ", ";\n        top: 0;\n        width: ", ";\n        background: none;\n      }\n\n      &:after {\n        width: ", ";\n        top: ", ";\n        height: ", ";\n        left: 12px;\n      }\n    }\n  \n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n  float: right;\n  margin-top: ", ";\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n  font-weight: ", ";\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n  font-size: ", ";\n  font-weight: ", ";\n  line-height: ", ";\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n  float: left;\n  margin-top: ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  display: inline;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  font-size: ", ";\n  font-weight: ", ";\n  line-height: ", ";\n\n  ", "\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  align-content: space-evenly;\n  display: flex;\n  flex-wrap: wrap;\n  margin-bottom: ", ";\n\n  span {\n    line-height: ", ";\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  display: inline;\n  float: right;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: inline-block;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  margin-bottom: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import Radio from '../FormElements/Radio/Radio';
import Ellipsis from '../Icon/Ellipsis';
import styled from 'styled-components';
import { rgba, rem } from 'polished';
import img from '../../static/icons/check-icon.svg';
import { Ellipse } from '../Icon/Ellipse';
import { MasterCard } from '../Icon';
import PropTypes from 'prop-types';

var DebitCard = function DebitCard(_ref) {
  var _ref$isDefault = _ref.isDefault,
      isDefault = _ref$isDefault === void 0 ? false : _ref$isDefault,
      className = _ref.className,
      cardNumber = _ref.cardNumber,
      expiryDate = _ref.expiryDate;
  return React.createElement(StyledDebitCard, {
    className: className
  }, React.createElement(CardTop, null, React.createElement(CardStatus, null, React.createElement(Radio, {
    defaultChecked: isDefault
  })), React.createElement(CardIcon, null, React.createElement(Ellipsis, null))), React.createElement(CardMid, null, React.createElement("span", null, React.createElement(Ellipse, null)), React.createElement("span", null, React.createElement(Ellipse, null)), React.createElement("span", null, React.createElement(Ellipse, null)), React.createElement(CardNumber, null, cardNumber)), React.createElement(CardBottom, null, React.createElement(CardExpiry, null, React.createElement(CardExpires, null, "Expires:"), ' ', React.createElement(CardExpiryDate, null, expiryDate)), React.createElement(CardType, null, React.createElement(MasterCard, null))));
};

var CardTop = styled.div(_templateObject(), rem('48px'));
var CardStatus = styled.div(_templateObject2());
var CardIcon = styled.div(_templateObject3());
var CardMid = styled.div(_templateObject4(), rem('18px'), rem('21px'));
var CardNumber = styled.span(_templateObject5(), rem('18px'), function (_ref2) {
  var theme = _ref2.theme;
  return theme.__OriginateLib.fontWeight.semiBold;
}, rem('28px'), ''
/* @include phone {
font-size: ${rem('14px')};
} */
);
var CardBottom = styled.div(_templateObject6());
var CardExpiry = styled.div(_templateObject7(), rem('45px'));
var CardExpires = styled.span(_templateObject8(), rem('14px'), function (_ref3) {
  var theme = _ref3.theme;
  return theme.__OriginateLib.fontWeight.semiBold;
}, rem('28px'));
var CardExpiryDate = styled.span(_templateObject9(), function (_ref4) {
  var theme = _ref4.theme;
  return theme.__OriginateLib.fontWeight.semiBold;
});
var CardType = styled.div(_templateObject10(), rem('45px'));
var StyledDebitCard = styled.div(_templateObject11(), function (_ref5) {
  var theme = _ref5.theme;
  return theme.__OriginateLib.colors.primary;
}, rem('1px'), rem('12px'), rem('2px'), rem('11px'), rem('22px'), rgba(0, 0, 0, 0.06), function (_ref6) {
  var theme = _ref6.theme;
  return theme.__OriginateLib.colors.white;
}, rem('186px'), rem('20px'), rem('345px'), ''
/* include phone {
width: 100%;
} */
, ''
/* .radio {
} */
, rem('2px'), function (_ref7) {
  var theme = _ref7.theme;
  return theme.__OriginateLib.colors.white;
}, function (_ref8) {
  var theme = _ref8.theme;
  return theme.__OriginateLib.colors.secondary;
}, img, rem('2px'), function (_ref9) {
  var theme = _ref9.theme;
  return theme.__OriginateLib.colors.white;
}, rem('0px'), rem('4px'), rem('4px'), function (_ref10) {
  var theme = _ref10.theme;
  return rgba(0, 0, 0, 0.02);
}, rem('25px'), rem('25px'), rem('21px'), rem('2px'), rem('21px'));
DebitCard.propTypes = {
  isDefault: PropTypes.bool,
  className: PropTypes.string,
  cardNumber: PropTypes.string,
  expiryDate: PropTypes.string
};
DebitCard.defaultProps = {
  className: '',
  isDefault: true,
  cardNumber: '',
  expiryDate: ''
};
export default DebitCard;
